### DISCUSSION TITLE
Background color setting for Desktop APP

### DISCUSSION THREAD

**Discussion by karasunouta** [ruffle-rs/ruffle#6219] [Ideas] [answered]
![TextInvisible](https://user-images.githubusercontent.com/188811/152653795-ad343e7b-b81c-491b-8d31-528befa1f68e.png)
Thank you for your brilliant APP!
I'm moved to tears for the rebirth of my favorite swf game with your desktop APP.

But there's just one difficulty.
This game's some messages are invisible for their background color seems black and text color is also black though some years ago I could read the texts quite well.

In my thought, the game's background color is none or transparent, and Ruffle's background color is black so this happens.
Is there any way to set this APP's background color white?

If there's not, I just eager it!
I would be overjoyed if the setting was add to this APP.
Thank you again.

**Comment by midgleyc** [comment]
Could you attach the SWF? I think the default color if unset is white.

Could it be that the original was displayed on a webpage configured with "bgcolor=white"?

**Comment by karasunouta** [comment]
Thank you for your response!
I'm not the author of the SWF, but for the purpose of its cultural value I'll share the file wishing his forgiveness.
[game032.zip](https://github.com/ruffle-rs/ruffle/files/8010399/game032.zip)

> Could it be that the original was displayed on a webpage configured with "bgcolor=white"?

It could be.
I'm sorry I'm quite not sure about it because several years have passed since I played the game on websites.

Only thing I can say is this game is extremely story-oriented one, and I was shocked yesterday as its very important messages are not shown...

If you verify the problem of this swf game, I'll be glad so much.

**Reply by midgleyc** [reply]
Thanks!

Here it is in Flash Player:
![image](https://user-images.githubusercontent.com/25565697/152687012-6de047da-9303-4c04-b56a-bc6c78096369.png)

So I think the image is supposed to be black, and the text white. The problem is likely https://github.com/ruffle-rs/ruffle/issues/1862, which prevents non-embedded fonts being shown, unless their characters are ASCII, which these Japanese characters are not.

**Comment by karasunouta** [comment]
AHHHH!
That's exactly what I was eager to see! Thank you!!
#1862 seems to be remaining open.
All I can do is to wait its progress? That's pity but I'm not a swf programmer...

**Comment by Lord-McSweeney** [comment, marked as answer]
#1862 has been resolved, at least on desktop.
